crates/sui-kvstore/src/writer.rs

```diff
@@ -139,56 +135,74 @@ where
                 .get_full_checkpoint_contents(&checkpoint_summary.content_digest)
                 .map_err(|_| anyhow!("Failed to read checkpoint content from store"))?
             {
-                let mut transactions = vec![];
-                let mut effects = vec![];
-                let mut events = vec![];
+                // let mut transactions = vec![];
+                // let mut effects = vec![];
+                // let mut events = vec![];
+                let mut objects = vec![];
 
                 for content in contents.iter() {
-                    let transaction_digest = content.transaction.digest().into_inner().to_vec();
-                    effects.push((transaction_digest.clone(), content.effects.clone()));
-                    transactions.push((transaction_digest, content.transaction.clone()));
-
-                    if let Some(event_digest) = content.effects.events_digest() {
-                        if let Some(tx_events) = store
-                            .get_transaction_events(event_digest)
-                            .map_err(|_| anyhow!("Failed to fetch events from the store"))?
-                        {
-                            events.push((event_digest.into_inner().to_vec(), tx_events));
-                        }
+                    let object_keys: Vec<_> = content
+                        .effects
+                        .all_changed_objects()
```

Comment:
Does it also make sense to include `all_deleted_objects()` from the effects? So, if i am reading the latest state of an object from kvstore i can tell it has been deleted.

Reply:
hm, not sure. Right now, it's a pure key-value table with the access pattern object_key => its historical version. Currently, there's no way to retrieve the latest version from just the kv store.

Reply:
cc @mystenmark: just double checking if `all_changed_objects` is the right collection to use here for backfill